docs/decisions/0009-moving-catalog-diffing-to-enterprise-catalog.rst

```diff
@@ -0,0 +1,89 @@
+stands now, the Integrated Channels assumes that the content metadata exporter will report the entirety of the
+customer's purchased content, then pass that payload to the content metadata transmitter which compares the payload with
+all saved `ContentMetadataItemTransmission` items under the customer to determine which items it (the transmitter) needs
+to create, update, and delete. It then chunks up each of the buckets, and sends those chunks (up to a limited number of
+times) to the client to send to the customer's LMS and saves the updated `ContentMetadataItemTransmission` entry.
+
+How the Integrated Channels could better determine if customer metadata needs updating
+======================================================================================
+
+proposed flow chart:
+
+.. image:: ../images/new-integrated-channels-metadata-flow.png
+  :width: 500
+
+If we build an endpoint that takes catalog UUIDs and a set of content keys linked to the uuid and returns three buckets
+of data:
+
+1) What content keys exist under the specified catalog that were not provided in the list of content keys
+
+2) What content keys don't exist under the specified catalog that were provided in the list of content keys
+
+3) What the last updated at times of the content keys are for the content keys provided that are under the specified
+catalog
+
+We would be able to move the responsibility of diffing what the customer has with what the customer should have to the
+enterprise catalog. This would mean that the only job of the Integrated Channels would be to compare the last updated
+times of the courses that already exist to get which courses need updates. Creates and deletes would already determined
+by the first endpoint.
+
+Further Improvements
+====================
+
+There are two additional improvements to explore with the Integrated Channels' metadata transmission flow. Firstly,
+we remove records from our transmission audit table when a delete request is issued. This can result in `lost` nodes of
+content (as we've actively seen from customers) where content can be assumed to be deleted on our end but have it exist
+still on the customer's external LMS. The fix here is pretty easy- if we choose to not delete records, but rather create
+a new, nullable `deleted_at` or equivalent field in the transmission record table. Then we would be able to mark and
+exclude any deleted records from appropriate look ups, but still have something to help us identify past courses that
+were sent to customers.
+
+As stated earlier, currently the only method to retrieve content under a customer's catalog is the singular, bulk
+`get_content_metadata` endpoint. This returns all content metadata belonging to an enterprise catalog. If we were to
+build out the existing endpoint where individual content key's worth of metadata could be specified in the request body,
```

Comment:
One more improvement to list -- being able to take an "after X" date value as a query parameter and return only those records which have been created/updated/deleted after that date.  We could consider a "before X" date value as well, which would enable additional scenarios such as a "between" type lookup.